AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -114,80 +110,71 @@ CAmSocketHandler::~CAmSocketHandler()
 void CAmSocketHandler::start_listenting()
 {
     mDispatchDone = false;
-    int16_t pollStatus;
 
 #ifndef WITH_TIMERFD 
     clock_gettime(CLOCK_MONOTONIC, &mStartTime);
 #endif    
     timespec buffertime;
     
-    VectorListPoll_t cloneListPoll;
-    VectorListPoll_t::iterator listmPollIt;
-    VectorListPollfd_t fdPollingArray; //!<the polling array for ppoll
-    
-    auto preparePollfd = [&](const sh_poll_s& row)
-    {
-        CAmSocketHandler::prepare((sh_poll_s&)row);
-        pollfd temp = row.pollfdValue;
-        temp.revents = 0;
-        fdPollingArray.push_back(temp);
-    };
+    VectorPollfd_t fdPollingArray; //!<the polling array for ppoll
 
     while (!mDispatchDone)
     {
-        if (mRecreatePollfds)
+        // iterate all times though list and update ppoll accordingly
+        auto fdPollIt = fdPollingArray.begin();
+        for (auto it = mMapShPoll.begin(); it != mMapShPoll.end(); )
         {
-            fdPollingArray.clear();
-            //freeze mListPoll by copying it - otherwise we get problems when we want to manipulate it during the next lines
-            cloneListPoll = mListPoll;
-            //there was a change in the setup, so we need to recreate the fdarray from the list
-            std::for_each(cloneListPoll.begin(), cloneListPoll.end(), preparePollfd);
-            mRecreatePollfds = false;
-        }
-        else
-        {
-            //first we go through the registered filedescriptors and check if someone needs preparation:
-            std::for_each(cloneListPoll.begin(), cloneListPoll.end(), CAmSocketHandler::prepare);
+            auto& elem = it->second;
+            switch (elem.state)
+            {
+                case poll_states_e::REMOVE:
+                    close(elem.pollfdValue.fd);
+                    fdPollIt = fdPollingArray.erase(fdPollIt);
+                    it = mMapShPoll.erase(it);
+                    continue;
+
+                case poll_states_e::UNINIT:
+                    fdPollIt = fdPollingArray.emplace(fdPollIt);
+                    // fallthrough
+
+                case poll_states_e::UPDATE:
+                    CAmSocketHandler::prepare(elem);
+                    elem.state = poll_states_e::VALID;
+                    *fdPollIt = elem.pollfdValue;
+                    break;
+
+                default:
```

Comment:
Can default ever happen?   Do you not make use of asserts / "fatal error" type handling, just for good measure?

Reply:
default is state VALID in our case.

Reply:
Oh.  That's unexpected.  Fine but then a comment would help to say this is an expected state, and we deliberately do nothing.  It's up to you of course.

Reply:
Added the VALID case and some more verbose comments.